wire/hybridbus.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wire
+
+import (
+	"context"
+	"fmt"
+
+	"polycry.pt/poly-go/errors"
+	"polycry.pt/poly-go/sync/atomic"
+)
+
+var _ Bus = (*hybridBus)(nil)
+
+// A Bus that can send messages over a local bus and a remote bus.
+type hybridBus struct {
+	buses []Bus
+}
+
+// NewHybridBus creates a hybrid bus that sends and receives messages over
+// multiple buses. Publishing a message over a hybrid bus will attempt to
+// publish it over all its sub-buses simultaneously. Subscribing a client to the
+// hybrid bus will subscribe it to all its sub-buses. The sub-buses can still be
+// accessed independently for more fine-grained control.
+func NewHybridBus(buses ...Bus) Bus {
```

Comment:
This only works for a certain kind of bus, right? It should be documented here which properties a bus must fulfill in order to be used with the hybrid bus.

Reply:
Added description to the constructor.